rero_ils/modules/notifications/dispatcher.py

```diff
@@ -25,27 +25,34 @@
 from ..locations.api import Location
 
 
-class Dispatcher():
+class Dispatcher:
     """Dispatcher notifications class."""
 
     def dispatch_notification(self, notification=None):
         """Dispatch the notification."""
+
+        def not_yet_implemented(*args):
+            """Do nothing placeholder for a notification."""
+            return
+
         if notification:
             data = notification.replace_pids_and_refs()
-            communication_channel = \
-                data['loan']['patron']['communication_channel']
-            if communication_channel == 'email':
-                self.send_mail(data=data)
-            elif communication_channel == 'sms':
-                pass
-            elif communication_channel == 'whatsapp':
-                pass
-            elif communication_channel == 'letter':
-                pass
+            communication_switcher = {
+                'email': Dispatcher.send_mail,
+                #  'sms': not_yet_implemented
+                #  'telepathy': self.madness_mind
+                #  ...
+            }
+            dispatcher_function = communication_switcher.get(
+                data['loan']['patron']['communication_channel'],
+                not_yet_implemented
```

Comment:
Why not raise an exception?

Reply:
Patron schema defined 4 communication channel (sms, whatsapp, ...). To not break this behavior and don't crash the server, I use a "do_nothing" function as used before using "pass" statement.